controllers/extensions/addon_controller_stages.go

```diff
@@ -1082,3 +1098,115 @@ func findDataKey[V string | []byte](data map[string]V, refObj extensionsv1alpha1
 	}
 	return false
 }
+
+func getKubeBlocksDeploy(ctx context.Context, r *AddonReconciler) (*v1.Deployment, error) {
+	deploys := &v1.DeploymentList{}
+	labelSelector := labels.SelectorFromSet(map[string]string{
+		constant.AppNameLabelKey:      constant.AppName,
+		constant.AppComponentLabelKey: "apps",
+	})
+	if err := r.Client.List(ctx, deploys, client.InNamespace(viper.GetString(constant.CfgKeyCtrlrMgrNS)),
+		client.MatchingLabelsSelector{Selector: labelSelector}); err != nil {
+		return nil, err
+	}
+	if deploys.Items == nil || len(deploys.Items) == 0 {
+		return nil, nil
+	}
+	if len(deploys.Items) > 1 {
+		return nil, fmt.Errorf("found multiple KubeBlocks deployments, please check your cluster")
+	}
+	return &deploys.Items[0], nil
+}
+
+func getKubeBlocksVersion(ctx context.Context, r *AddonReconciler) (string, error) {
+	deploy, err := getKubeBlocksDeploy(ctx, r)
+	if err != nil || deploy == nil {
```

Comment:
give some messages to user

Reply:
here I change the code, which deleted the judgement of deploy == nil, so if the kubeblocks deployment is nil, the corresponding error will be return